### ISSUE TITLE
false positive unused require in keys destructuring

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#273]
`echo '(ns f (:require [clojure.string :as s])) (defn f [{::s/keys [x]}] x)' | clj-kondo --lint -`
will report
`<stdin>:1:18: warning: namespace clojure.string is required but never used`

cc @souenzzo

**Comment by souenzzo** [comment]
Not sure if it's a regression but
```
~ echo '(ns f (:require [clojure.string :as s])) :s/foo'  | clj -A:kondo --lint - 
linting took 68ms, errors: 0, warnings: 0
```
I'm using git deps with sha
`a2bef25ac34cde1d3a903d80aa827455d2ee6a38`

**Comment by borkdude** [comment]
@souenzzo Existing bug apparently, thanks for spotting it. Will fix.

```
echo '(ns f (:require [clojure.string :as s])) :s/foo'  | /usr/local/bin/clj-kondo --lint -
linting took 33ms, errors: 0, warnings: 0
```

**Comment by souenzzo** [comment]
But yeah, about this bug, it's fixed.  :)
